articles/solving-linear-equations-using-matlab/index.md

```diff
@@ -0,0 +1,211 @@
+```matlab
+function [x0 y0] = intersectPoints(m1,m2,b1,b2) %m is the gradient while b is the constant. 
+x0 = (b2-b1)/(m1-m2); %find the x point
+y0 = m1*x0+b1;
+end
+```
+To call this function in the command window;
+```matlab
+intersectPoints(-4/5, 3/2, 6/5, -14/2) 
+```
+the output will be
+```
+ans =
+
+    3.5652
+```
+This function gives a single output corresponding to the `x` value for the point of intersection. To find the `y`, you can replace the `x` in the equation and get your `y`.
+The diagram shows that the solution is similar to that we found before(3.5652,-1.6522). It means that the solution is correct. Now assume that now we have the equation below;
+$$
+\begin{cases} 4x + 5y=6\\8x + 10y=12\\ -2x -2.5y =-3\\ 7x - y =25\end{cases}
+$$
+Let's plot the line of the three equations. To do that, we add the below code;
+```Matlab
+y3 = 7*x - 25;
+hold on
+plot(x, y1, x,y2, x, y3)
+```
+![output](/engineering-education/solving-linear-equation-using-matlab/linear_three.png)
+When you execute the above commands, we get a 3rd line which is for the third equation. Looking at the plots, you may think these three lines intersect at a common point, but this is not the case. To see this, zoom out the plots. To zoom out, click on the magnifying lens icon having `+` inside it. After clicking it, move to the point at which the lines seems to intersect a few times, and the result will be;
+![output](/engineering-education/solving-linear-equation-using-matlab/linear_four.png)
+This shows that there is no perfect solution to these problems. If we try solving it using matlab and plot this output to see the position that matlab gives as the output. This can be done by the code below;
+```matlab
+A = [4 5; 3 -2; 7 -1];
+b = [6; 14;25];
+x = A\b;
+hold on
+plot(x(1),x(2),'k*')
+plot(x(1),x(2),'r*')
+```
+![output](/engineering-education/solving-linear-equation-using-matlab/linear_five.png)
+As you can see, Matlab is trying to locate a point close to all three lines, and this is an approximation. The error to the solution is found by;
+```Matlab
+error = A*x -b
+```
+Matlab locates this closest point by finding the square root of the sum of squares of elements of the input vector. Locating this point is by use of the `norm` function.
+```Matlab
+norm(error)
+```
+The output is `0.7941` and is known as `optimum solution in the least square sense`. It is the point at which the `norm(error)` is as small as possible. If we try other points, we get `norm(error)` to be higher than `0.7941`.
+
+### Conclusion
+Solving simultaneous equations in Matlab depends on the type of problem that you are handling. As a user, you should know the type of equation or problem you are trying to solve. In addition, it will help you know the output you expect from Matlab. What you must know is that Matlab will provide a solution to all these problems. Also, knowing the type of equation helps you avoid errors in linear equations.
```

Comment:
```suggestion
Solving simultaneous equations in Matlab depends on the type of problem that you are handling. As a user, you should know the type of equation or problem you are trying to solve. Besides, it will help you know the output you expect from Matlab. What you must know is that Matlab will provide a solution to all these problems. Also, knowing the type of equation helps you avoid errors in linear equations.
```